### ISSUE TITLE
SDWebimage can't get the image

### ISSUE THREAD

**Issue by forestfsl** [rs/SDWebImage#1831]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Is such a situation：
Our company has a pre-release account，But when loading the app has not yet boarded，
This is a normal case because there is no url to get the image，
But after I use the pre-release account login, re-request and refresh the tableView, the picture is still not normal display，
After exiting the app, re-enter, the page is displayed normally，But those who first get the picture url is not the case, which is why?
I try to see the source code, but in the dispatch_main_async_safe this method inside the break point, breakpoint did not enter, ask the author to help，thanks

**Comment by mythodeia** [comment]
can you replicate this in the demo project?

**Comment by mythodeia** [comment]
closing this. please reopen when you have more info. Also please try it with the latest version of the library.

**Comment by bpoplauschi** [comment]
Hmm, maybe the image server returns an error the 1st time and we might added it to failedURLs, so we are not trying again. Might worth looking into that
